graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_numerical_transformation.py

```diff
@@ -0,0 +1,349 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import logging
+from typing import Optional, Sequence
+
+from pyspark.sql import DataFrame
+from pyspark.sql import functions as F
+from pyspark.sql.types import ArrayType, FloatType
+from pyspark.ml.feature import MinMaxScaler, Imputer, VectorAssembler, ElementwiseProduct
+from pyspark.ml.linalg import DenseVector
+from pyspark.ml.stat import Summarizer
+from pyspark.ml import Pipeline
+from pyspark.ml.functions import array_to_vector, vector_to_array
+
+import numpy as np
+
+from graphstorm_processing.constants import SPECIAL_CHARACTERS
+from .base_dist_transformation import DistributedTransformation
+from ..spark_utils import rename_multiple_cols
+
+
+def apply_imputation(cols: Sequence[str], shared_imputation: str, input_df: DataFrame) -> DataFrame:
+    """
+    Applies a single imputation to input dataframe, individually to each of the columns
+    provided in the cols argument.
+    """
+    assert shared_imputation in [
+        "mean",
+        "median",
+        "mode",
```

Comment:
It's used in stats to indicate the most frequent value in a dataset: https://en.wikipedia.org/wiki/Mode_(statistics)

Reply:
Can you add https://en.wikipedia.org/wiki/Mode_(statistics) as a comment here?

Reply:
Will do. Just a note that `mode` is only needed by the Spark transformer internally, we expose `most_frequent` to the user. I'll modify this function to accept both, since it's possible other callers might exist.